### PR TITLE
Audio input implementation and fixes

### PR DESCRIPTION
**Pull request by hexkyz**

### CODE REVIEW

File: nx/include/switch/audio/audio.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * @author hexkyz
+ * @copyright libnx Authors
+ */
+#pragma once
+
+#include "../types.h"
+
+typedef enum {
+    PcmFormat_Invalid = 0,
+    PcmFormat_INT8 = 1,
```
Comment (plutooo) [inline nx/include/switch/audio/audio.h]: Int8
Int16
..
Float

File: nx/include/switch/services/audout.h [hunk 2]
```diff
@@ -53,14 +43,14 @@ Result audoutContainsAudioOutBuffer(AudioOutBuffer *Buffer, bool *ContainsBuffer
  * @param source AudioOutBuffer containing the source sample data to be played.
  * @param released AudioOutBuffer to receive the last played buffer.
  */
-Result audoutPlayBuffer(AudioOutBuffer *source, AudioOutBuffer *released);
+Result audoutPlayBuffer(AudioOutBuffer *source, AudioOutBuffer **released);
 
 /**
  * @brief Waits for audio playback to finish.
  * @param released AudioOutBuffer to receive the last played buffer.
  * @param timeout Timeout value, use U64_MAX to wait until all finished.
```
Comment (yellows8) [inline nx/include/switch/services/audout.h]: Comment for released_count missing.

File: nx/include/switch/services/audin.h [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+ * @brief Submits an audio sample data buffer for capturing and waits for it to finish capturing.
+ * @brief Uses \ref audinAppendAudioInBuffer and \ref audinWaitCaptureFinish internally.
+ * @param source AudioInBuffer containing the buffer to hold the captured sample data.
+ * @param released AudioInBuffer to receive the last captured buffer.
+ */
+Result audinCaptureBuffer(AudioInBuffer *source, AudioInBuffer **released);
+
+/**
+ * @brief Waits for audio capture to finish.
+ * @param released AudioInBuffer to receive the last captured buffer.
```
Comment (yellows8) [inline nx/include/switch/services/audin.h]: Comment for released_count missing.

**Comment by yellows8** [comment]
aud*.h "Supported PCM format (INT16)." <- Sync with audio.h.

**Comment by Cpasjuste** [comment]
This work fine and can be merged, thanks @hexkyz

**Pull request merged**